WIP EXPERIMENT DO NOT MERGE

chrischdi opened the pull request.

CatherineF-dev committed:
daemonset-sharding-example

rexagod committed:
Handle unit length `valueFrom` values

Handle unit length `valueFrom` values and skip strings where we expect
them to be type-cast-able to `float64`, instead of erroring, since that
is the expected behavior, and what's being done for other types.

rexagod committed:
Add support for variable VKs in CRS config

Add support for variable VKs in CRS config, while maintaining a cache
of discovered GVKs in the cluster, and updating it every 30s.

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

pwschuurman committed:
Add StatefulSet Start Ordinal metrics for KEP-3335

jabdoa2 committed:
support "True" and "False" as string in custom-resource-state for operator status conditions

jabdoa2 committed:
lowercase string and simplify logic

Co-authored-by: Manuel Rüger <manuel@rueg.eu>

jabdoa2 committed:
Document how gauges convert types to float

jabdoa2 committed:
Improve formating

mrueg committed:
Use structured logging

mrueg committed:
go.mod: Update dependencies

mrueg committed:
*: Cut v2.8.0

chrischdi committed:
Implement metric-gen tool

Implements the metric-gen tool which could get used to create custom resource
configurations directly from code, similar to what controller-gen does.

Commit:
Fixing emitting of ready time metrics when condition is false

k8s-ci-robot committed:
Merge pull request #1965 from mrueg/convert-structure-logging

Use structured logging

rexagod committed:
fixup! Handle unit length `valueFrom` values

k8s-ci-robot committed:
Merge pull request #1963 from jabdoa2/support_string_bool_in_custom_resource_state

support "True" and "False" as string in custom-resource-state

mrueg committed:
Set OpenMetrics content header

See: https://github.com/prometheus/common/blob/main/expfmt/encode.go#L86

See: https://github.com/kubernetes/kube-state-metrics/issues/1973

CatherineF-dev committed:
Migrate all NewFamilyGenerator to NewFamilyGeneratorWithStability

k8s-ci-robot committed:
Merge pull request #1976 from CatherineF-dev/force-adding-metric-stability

Migrate all NewFamilyGenerator to NewFamilyGeneratorWithStability

k8s-ci-robot committed:
Merge pull request #1971 from ryanrolds/rolds/bug_fix_pod_container_ready_time

Fixing emitting of ready time metrics when condition is false

CatherineF-dev committed:
Clean up NewFamilyGenerator in comments

k8s-ci-robot committed:
Merge pull request #1977 from CatherineF-dev/clean-up-NewFamilyGenerator

Clean up function NewFamilyGenerator in comments

k8s-ci-robot committed:
Merge pull request #1974 from mrueg/openmetrics

Set OpenMetrics content header

mrueg committed:
Update docs/customresourcestate-metrics.md

mrueg committed:
Update docs/customresourcestate-metrics.md

k8s-ci-robot committed:
Merge pull request #1964 from jabdoa2/patch-1

Document how gauges convert types to float

k8s-ci-robot committed:
Merge pull request #1958 from rexagod/1947

Handle unit length `valueFrom` values

k8s-ci-robot committed:
Merge pull request #1967 from mrueg/prep-v2.8.0

Prepare v2.8.0

k8s-ci-robot committed:
Merge pull request #1986 from kubernetes/release-2.8

Merge Release 2.8.0 back to main

mrueg committed:
feat!: Remove VerticalPodAutoscaler

BREAKING CHANGE: VPA is now supported only via Custom Resource Metrics

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

mrueg committed:
feat(CustomResourceState): Support quantities

mrueg committed:
feat: Enable semantic commit messages

mrueg committed:
chore: Update CONTRIBUTING.md

Co-authored-by: Mario Constanti <github@constanti.de>

sbueringer commented on go.mod:
```diff
@@ -57,19 +58,20 @@ require (
 	github.com/modern-go/reflect2 v1.0.2 // indirect
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 // indirect
 	github.com/mwitkow/go-conntrack v0.0.0-20190716064945-2f068394615f // indirect
+	github.com/onsi/ginkgo/v2 v2.6.1 // indirect
+	github.com/onsi/gomega v1.24.2 // indirect
```
I wonder why those dependencies got added
There seem to be no changes in the main go modules explaining this

sbueringer commented on exp/metric-gen/metrics.yaml:
```diff
@@ -0,0 +1,13 @@
+---
```
Should this be checked in? (or just tmp file from testing?)

sbueringer commented on exp/metric-gen/main.go:
```diff
@@ -0,0 +1,229 @@
+)
+
+func init() {
+	for genName, gen := range allGenerators {
```
In my opinion the controller-gen way of handling input flags is totally overengineered for our use case (assuming we just need a few flags).

I would try to get rid of all of this complexity

I ~ did this with the existing crd generator here: https://github.com/kubernetes-sigs/cluster-api/blob/296224d73b33a3619e681c6c87a2dc3c694de401/hack/tools/runtime-openapi-gen-2/main.go#L80-L92

(the code is not very cleaned up, but it might show that you don't need most of the stuff your using here like genall, ...)

Do you need cobra or just a few flags? then the go flag package might be the easiest way.

sbueringer commented on exp/metric-gen/metric/parser.go:
```diff
@@ -0,0 +1,247 @@
+	}
+}
+
+func (p *Parser) NeedResourceFor(groupKind schema.GroupKind) {
```
I think we'll need a lot of godoc explaining what all this is doing to make it maintainable for folks who are not super-famiilar with the controller-gen implementation

clamoriniere committed:
fix: public Builder compatibility with the BuilderInterface

clamoriniere committed:
fix: public Builder compatibility with the BuilderInterface

k8s-ci-robot committed:
Merge pull request #1994 from clamoriniere/fix-builder-interface

fix: public Builder compatibility with the BuilderInterface

edwardmlyte committed:
Correct link for customising kube prometheus

Original page moved location.

rexagod committed:
Don't crash on non-existent path values

Don't crash on non-existent path values in CRS.

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

rexagod committed:
Don't crash on non-existent path values

Don't crash on non-existent path values in CRS.

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

rexagod committed:
fixup! Don't crash on non-existent path values

rexagod committed:
fixup! Don't crash on non-existent path values

k8s-ci-robot committed:
Merge pull request #1998 from rexagod/1992

Don't crash on non-existent path values

k8s-ci-robot committed:
Merge pull request #2000 from edwardmlyte/main

Correct link for customising kube prometheus

mrueg committed:
build: Bump to go 1.19.6

mrueg committed:
build: Bump to go 1.19.6

k8s-ci-robot committed:
Merge pull request #2001 from mrueg/go-1.19.6

build: Bump to go 1.19.6

mrueg committed:
*: Cut v2.8.1

mrueg committed:
build: Build with go 1.20.2

Also bump prometheus and golang dependencies

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

k8s-ci-robot committed:
Merge pull request #2002 from mrueg/prep-v2.8.1

Prepare bug fix release v2.8.1

rexagod committed:
Adhere to OTel-Prometheus standard for labels

Adhere to OTel-Prometheus standard for generated labels in CRS.

Ref.: https://github.com/open-telemetry/opentelemetry-specification/blob/8946dfc6a2302f78b0224fcc3f4dfb816a7bb1f4/specification/compatibility/prometheus_and_openmetrics.md?plain=1#L224-L229
Additional info.: OSM has a hardcoded approach for doing this right now: https://github.com/openshift/openshift-state-metrics/blob/master/pkg/collectors/utils.go#L29

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

rexagod committed:
Fix label resolution for maps

The `map[string]interface{}` resolution searched for values within various nested fields, but not the relative path itself.

Ref.: https://github.com/kubernetes/kube-state-metrics/blob/ca1da6b22ed28fe3aa6cac98705cff71c614eac0/pkg/customresourcestate/registry_factory.go#L603

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

rexagod committed:
Add `rexagod` to approvers

Based on the nomination proposed on the Slack channel (thanks, Manuel!).

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

k8s-ci-robot committed:
Merge pull request #2008 from rexagod/add-rexagod-as-approver

Add `rexagod` to approvers

k8s-ci-robot committed:
Merge pull request #2007 from rexagod/fix-label-resolution-maps

Fix label resolution for maps

pwschuurman committed:
Add gauge implementation of kube_statefulset_ordinals_start

dfredell committed:
Add cli refrences to the docs around annotations and labels

mrueg committed:
feat(customresourcestate): Support percentages

k8s-ci-robot committed:
Merge pull request #1989 from mrueg/crs-support-quantity

feat(CustomResourceState): Support quantities and percentages

Commit:
Clarify pod status in useful queries

Signed-off-by: Artur Rodrigues <artur.rodrigues@lacework.net>

pwschuurman committed:
Fix indentation formatting for test WANT stanza in statefulset_test.go

arturhoo committed:
Update golangci-lint version

k8s-ci-robot committed:
Merge pull request #2021 from arturhoo/bump-golangci-lint-version

Update golangci-lint version to 1.51.2

CatherineF-dev committed:
Only use OpenMetrics and Text in contentType

k8s-ci-robot committed:
Merge pull request #1959 from pwschuurman/kep-3335-beta-metrics

Add StatefulSet Start Ordinal metrics for KEP-3335

k8s-ci-robot committed:
Merge pull request #2003 from mrueg/go-1.20

build: Build with go 1.20.1

CatherineF-dev committed:
Update pkg/metricshandler/metrics_handler.go

Co-authored-by: Manuel Rüger <manuel@rueg.eu>

k8s-ci-robot committed:
Merge pull request #2018 from arturhoo/patch-1

Improve wording for Prometheus rule example in pod-metrics

k8s-ci-robot committed:
Merge pull request #2024 from CatherineF-dev/set-contentType

Only use OpenMetrics and Text in contentType

mrueg committed:
*: Cut v2.8.2

mrueg committed:
doc: Describe the process to become a maintainer

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

k8s-ci-robot committed:
Merge pull request #2030 from mrueg/add-maintainers

doc: Describe the process to become a maintainer

k8s-ci-robot committed:
Merge pull request #2017 from mrueg/rm-vpa

feat!: Remove VerticalPodAutoscaler

mrueg committed:
feat: Use exporter-toolkit landingpage

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

k8s-ci-robot committed:
Merge pull request #2033 from mrueg/merge-2.8.2-to-main

Update changelog for v2.8.2 in main

k8s-ci-robot committed:
Merge pull request #1873 from CatherineF-dev/daemonset-sharding-example

docs: Daemonset sharding example

mrueg committed:
fix: Use correct writer

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

k8s-ci-robot committed:
Merge pull request #2037 from mrueg/fix-2036

fix: Use correct writer

mrueg committed:
.github: Use actions/setup-go@v4

https://github.com/actions/setup-go/releases/tag/v4.0.0
Signed-off-by: Manuel Rüger <manuel@rueg.eu>

k8s-ci-robot committed:
Merge pull request #2038 from mrueg/gh-checkout-gov4

.github: Use actions/setup-go@v4

k8s-ci-robot committed:
Merge pull request #2034 from mrueg/landingpage

feat: Use exporter-toolkit landingpage

jcpunk committed:
Add parameters for PodSecurity restricted

Signed-off-by: Pat Riehecky <riehecky@fnal.gov>

k8s-ci-robot committed:
Merge pull request #2042 from jcpunk/pss-restricted

Add parameters for PodSecurity restricted

juanjjaramillo committed:
Ignore casing while scraping failure reason for `kube_job_status_failed`

juanjjaramillo committed:
Map exact value provided but update value to match the correct casing

k8s-ci-robot committed:
Merge pull request #2046 from juanjjaramillo/juanjjaramillo/fix_job_failure_reason

Fix casing while scraping failure reason for `kube_job_status_failed`

murphd40 committed:
2047: Add LabelsFromPath functionality to Info metrics

k8s-ci-robot committed:
Merge pull request #2004 from rexagod/fix-star-path-labels

Adhere to OTel-Prometheus standard for labels

k8s-ci-robot committed:
Merge pull request #2048 from murphd40/2047

2047: Add LabelsFromPath functionality to Info metrics

RamakrishnanArun committed:
Move to 1.20.4

jkroepke committed:
Implement multi-cluster alerts

k8s-ci-robot committed:
Merge pull request #2056 from RamakrishnanArun/cvefix/1.20.3

Update go version to 1.20.4

mrueg committed:
Update dependencies

k8s-ci-robot committed:
Merge pull request #2061 from mrueg/update-deps-05

Update dependencies

askulkarni2 committed:
Use dl.k8s.io instead of hardcoded GCS URIs

askulkarni2 committed:
add redirect to the curl command

k8s-ci-robot committed:
Merge pull request #2063 from askulkarni2/chore/deprecate-kubernetes-release-url

Use dl.k8s.io instead of kubernetes-release bucket

k8s-ci-robot committed:
Merge pull request #2058 from jkroepke/multi-cluster

Implement multi-cluster alerts

rexagod committed:
Allow field KV general matching

Currently all matching support is limited to map-list representations;
this adds support for a more general "key:value" based matching
expression for any `interface{}` field (non-map-list-like).

CatherineF-dev committed:
Update README on listing multi architecture images

k8s-ci-robot committed:
Merge pull request #2072 from CatherineF-dev/patch-1

Update README on listing multi architecture images

k8s-ci-robot committed:
Merge pull request #1851 from rexagod/1848

Allow optional VK in CR metrics

a-hilaly committed:
Add namespace label to `kube_lease_renew_time`

This patch adds the namespace label to the `kube_lease_renew_time` and
updates unit tests to validate the changes.

Fixes github.com/kubernetes/kube-state-metrics

k8s-ci-robot committed:
Merge pull request #2073 from A-Hilaly/lease-renew-time-labels

Add namespace label to `kube_lease_renew_time`

maximee-leroyy committed:
feat(pvc): support kube_persistentvolumeclaim_deletion_timestamp

Signed-off-by: Maxime Leroy <19607336+maxime1907@users.noreply.github.com>

maximee-leroyy committed:
feat(pv): support kube_persistentvolume_deletion_timestamp

Signed-off-by: Maxime Leroy <19607336+maxime1907@users.noreply.github.com>

mrueg committed:
chore: Bump dependencies for next release

k8s-ci-robot committed:
Merge pull request #2076 from mrueg/bump-deps-for-release

chore: Bump dependencies for next release

rexagod committed:
*: Cut v2.9.0 release

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

k8s-ci-robot committed:
Merge pull request #2077 from rexagod/prep-v2.9.0

*: Cut v2.9.0 release

mrueg committed:
Increase timeout and update cloudbuild image

k8s-ci-robot committed:
Merge pull request #2083 from mrueg/cloudbuild

Increase timeout and update cloudbuild image

rexagod committed:
*: Cut v2.9.2

k8s-ci-robot committed:
Merge pull request #2084 from rexagod/prep-2.9.2

*: Cut v2.9.2

k8s-ci-robot committed:
Merge pull request #2082 from kubernetes/release-2.9

Sync `main` with `release-2.9`

mrueg committed:
chore: Bump to kubernetes 1.27.x

k8s-ci-robot committed:
Merge pull request #2087 from mrueg/k8s-1.27

go.mod: Bump to k8s 1.27.x

chalapat committed:
Fix cyclo complexity in horizontalpodautoscaler file function is hpaMetricFamilies

k8s-ci-robot committed:
Merge pull request #2089 from venkatbvc/gocylo_incremental_fix

Fix cyclomatic complexity in KSM files

swarup-stripe committed:
add new metric kube_pod_service_account

k8s-ci-robot committed:
Merge pull request #1990 from mrueg/add-semantic-pr

feat: Enable semantic commit messages

CatherineF-dev committed:
Update multi-arch version to v2.9.2 in README.md

k8s-ci-robot committed:
Merge pull request #2101 from CatherineF-dev/patch-2

docs: update multi-arch version to v2.9.2 in README.md

mrueg committed:
chore: Update dependencies

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

mrueg committed:
docs: Document RBAC and remove outdated info

Adds a paragraph on RBAC and removes the outdated information on setting
the customresource in the --resources flag.

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

k8s-ci-robot committed:
Merge pull request #2106 from mrueg/doc-crs

docs: Document RBAC and remove outdated info

mrueg committed:
chore: Fix lint

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

mrueg committed:
build: Reorder CI steps to use go mod cache

dgrisonnet committed:
*: add webhooks client config service metrics

In order to better identify, prioritize, and debug webhook latency
issues it is important to have a metric that would point to the resource
it is responsible for.  However, it is not possible to have that
dimension in the metrics exposed by Kubernetes because of the unbound
cardinality that such a label would have.

The name of the webhook could be an alternative since it usually
contains some information about the resource that the webhook targets,
however this is not very practical to use in multi-tenants
environments.

A solution for these kind of platform is to tie a specific webhook to a
namespace in order to be able to know which tenant manages it and take
actions depending on that. This is achieveable by leveraging the client
config information of webhooks configured via WebhookConfiguration
resources since Services are namespaced objects.

With these new metrics, users will be able to split the alerting
severity of webhook latency / rejection rate per namespace on top of
being able to do it based on the webhook name. This is key in
environment where administrators don't have control over the webhooks
installed by the various tenants.

Signed-off-by: Damien Grisonnet <dgrisonn@redhat.com>

mrueg committed:
Fix govulncheck

k8s-ci-robot committed:
Merge pull request #2112 from mrueg/use-caching

build: Reorder CI steps to use go mod cache

k8s-ci-robot committed:
Merge pull request #2105 from mrueg/update-deps-0623

chore: Update dependencies

ntoofu committed:
Add filtering for Lease metrics

fpetkovski committed:
chore: Remove myself from maintainers

I will be removing myself from the maintainers role due to
available time to work on KSM.

Signed-off-by: fpetkovski <filip.petkovsky@gmail.com>

k8s-ci-robot committed:
Merge pull request #2126 from fpetkovski/remove-fpetkovski-maintainer

chore: Remove myself from maintainers

mrueg committed:
chore: Sync security contacts with approvers

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

mrueg committed:
chore: Add dependabot config

To check and update go.mod and github actions on a weekly base

k8s-ci-robot committed:
Merge pull request #2122 from ntoofu/fix-2121

fix: Add filtering for Lease metrics

k8s-ci-robot committed:
Merge pull request #2114 from dgrisonnet/webhook-services

feat: Add webhooks client config service metrics

k8s-ci-robot committed:
Merge pull request #2074 from maxime1907/featpvcdeletiontimestamp

feat(pvc): support kube_persistentvolumeclaim_deletion_timestamp

k8s-ci-robot committed:
Merge pull request #2075 from maxime1907/featpvdeletiontimestamp

feat(pv): support kube_persistentvolume_deletion_timestamp

k8s-ci-robot committed:
Merge pull request #2096 from swarup-stripe/add-SA-to-kube-pod-info

feat: Adds new metric `kube_pod_service_account`

k8s-ci-robot committed:
Merge pull request #2127 from mrueg/update-security-contacts-v2

chore: Sync security contacts with approvers

machadovilaca committed:
feat: add volumemode to PVC info metric

Signed-off-by: João Vilaça <jvilaca@redhat.com>

k8s-ci-robot committed:
Merge pull request #2128 from mrueg/add-dependabot

chore: Add dependabot config

dependabot[bot] committed:
build(deps): Bump k8s.io/component-base from 0.27.3 to 0.27.4

Bumps [k8s.io/component-base](https://github.com/kubernetes/component-base) from 0.27.3 to 0.27.4.
- [Commits](https://github.com/kubernetes/component-base/compare/v0.27.3...v0.27.4)

---
updated-dependencies:
- dependency-name: k8s.io/component-base
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #2136 from kubernetes/dependabot/go_modules/k8s.io/component-base-0.27.4

build(deps): Bump k8s.io/component-base from 0.27.3 to 0.27.4

dependabot[bot] committed:
build(deps): Bump k8s.io/sample-controller from 0.27.3 to 0.27.4

Bumps [k8s.io/sample-controller](https://github.com/kubernetes/sample-controller) from 0.27.3 to 0.27.4.
- [Commits](https://github.com/kubernetes/sample-controller/compare/v0.27.3...v0.27.4)

---
updated-dependencies:
- dependency-name: k8s.io/sample-controller
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #2139 from kubernetes/dependabot/go_modules/k8s.io/sample-controller-0.27.4

build(deps): Bump k8s.io/sample-controller from 0.27.3 to 0.27.4

chihshenghuang committed:
fix non-existent valueFrom path value throw panic error issue

opeco17 committed:
feat: hot reload kubeconfig

opeco17 committed:
feat: disable labels and annotations metrics when metric-annotations-allowlist and metric-labels-allowlist are not provided

machadovilaca committed:
Update docs

Signed-off-by: João Vilaça <jvilaca@redhat.com>

opeco17 committed:
feat: implement kube_pod_status_initialized_time

k8s-ci-robot committed:
Merge pull request #2134 from machadovilaca/add-volume-mod-to-pvc-info-metric

feat: add volumemode to PVC info metric

david-martin committed:
Add example for non map Arrays

chrischdi committed:
feat(custommetrics) always extract the metric headers but only write them when we have metrics

chrischdi committed:
feat(customresource) log no _info suffix in name only once per reading the configuration

k8s-ci-robot committed:
Merge pull request #2157 from chrischdi/pr-info-no-suffix-spam

fix: custommetrics: log no _info suffix in name only once per reading the configuration

k8s-ci-robot committed:
Merge pull request #2151 from david-martin/non-map-array-example

docs:  Add example for non map Arrays

k8s-ci-robot committed:
Merge pull request #2140 from chihshenghuang/fix-guage-value-func-panic

fix: Don't crash on non-existent valueFrom path values

dfredell committed:
Merge remote-tracking branch 'upstream/main'

Commit:
fix(customresourcestate): typo in Error message

Signed-off-by: Adrian Berger <adrian.berger@bedag.ch>

k8s-ci-robot committed:
Merge pull request #2163 from adberger/main

fix(customresourcestate): typo in Error message

k8s-ci-robot committed:
Merge pull request #2148 from opeco17/feature/implement-pod-initialized-time

feat: implement kube_pod_status_initialized_time

k8s-ci-robot committed:
Merge pull request #2145 from opeco17/feature/disable-labels-annotations-metrics-by-default

feat: disable labels and annotations metrics when metric-annotations-…

dgrisonnet committed:
options: update labels/annotations allowlist doc

Signed-off-by: Damien Grisonnet <dgrisonn@redhat.com>

k8s-ci-robot committed:
Merge pull request #2164 from dgrisonnet/allowlist-doc

docs: Update labels and annotations allowlist

mrueg committed:
fix: Index out of range in metrics_store.SanitizeHeaders

2023-08-24T18:08:21.796778057Z 2023/08/24 18:08:21 http: panic serving ip:port: runtime error: index out of range [0] with length 0
2023-08-24T18:08:21.796814387Z goroutine 589687 [running]:
2023-08-24T18:08:21.796823334Z net/http.(*conn).serve.func1()
2023-08-24T18:08:21.796830520Z  /usr/local/go/src/net/http/server.go:1854 +0xbf
2023-08-24T18:08:21.796838117Z panic({0x19c2060, 0xc0043060a8})
2023-08-24T18:08:21.796844764Z  /usr/local/go/src/runtime/panic.go:890 +0x263
2023-08-24T18:08:21.796852064Z k8s.io/kube-state-metrics/v2/pkg/metrics_store.SanitizeHeaders(...)
2023-08-24T18:08:21.796858974Z  /root/go/src/k8s.io/kube-state-metrics/pkg/metrics_store/metrics_writer.go:94
2023-08-24T18:08:21.796866007Z k8s.io/kube-state-metrics/v2/pkg/metricshandler.(*MetricsHandler).ServeHTTP(0xc000221490, {0x1d425a0, 0xc005872700}, 0xc003f9d300)
2023-08-24T18:08:21.796873054Z  /root/go/src/k8s.io/kube-state-metrics/pkg/metricshandler/metrics_handler.go:211 +0x8fe
...

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

mrueg committed:
build: Bump dependencies

mrueg committed:
build: Fix lint

k8s-ci-robot committed:
Merge pull request #2166 from mrueg/fix-panic

fix: Index out of range in metrics_store.SanitizeHeaders

opeco17 committed:
feat: enable metric-annotations-allowlist and metric-labels-allowlist for ResourceQuota

k8s-ci-robot committed:
Merge pull request #2175 from opeco17/feature/allow-annotations-and-labels-for-resource-quota

feat: enable metric-annotations-allowlist and metric-labels-allowlist for ResourceQuota

k8s-ci-robot committed:
Merge pull request #2168 from mrueg/bump-deps

build: Bump dependencies

k8s-ci-robot committed:
Merge pull request #2154 from chrischdi/pr-custom-metrics-missing-header

fix: custommetrics: always extract the headers but only write it when we have metrics

mrueg committed:
chore: Cut v2.10.0

Co-authored-by: Damien Grisonnet <damien.grisonnet@epita.fr>
Signed-off-by: Manuel Rüger <manuel@rueg.eu>

dfredell committed:
Run all docs through a markdown table formatter so they are more easily read in their text form. Used Markdown All in One in VSCode.

dfredell committed:
Merge remote-tracking branch 'upstream/main'

dfredell committed:
Add Description to all doc markdown tables. Add controlled by cli note about labels and annotations to all markdown files.

k8s-ci-robot committed:
Merge pull request #2067 from rexagod/colon-general-matching

feat: allow field KV general matching

k8s-ci-robot committed:
Merge pull request #2144 from opeco17/feature/kubeconfig-hot-reload

feat: hot reload kubeconfig

k8s-ci-robot committed:
Merge pull request #1988 from dfredell/main

docs: Add cli refrences to the docs around annotations and labels

chrischdi committed:
Implement unit and integration tests

MarkSRobinson committed:
feat: Add support for endpoint topology routing hints (#2090)

* Add support for endpoint topology routing hints

Fix goimports

Fix gocritic issue

* Update docs/endpointslice-metrics.md

Co-authored-by: Manuel Rüger <manuel@rueg.eu>

* Update internal/store/endpointslice.go

Co-authored-by: Manuel Rüger <manuel@rueg.eu>

* Update Dockerfile

* Update Dockerfile

* Update endpointslice_test.go

* Redesign metric

* Update endpointslice.go

Remove unneeded label

* Update endpointslice.go

* Update endpointslice-metrics.md

* Fix test

* Remove hostname

* Update docs/endpointslice-metrics.md

---------

Co-authored-by: Manuel Rüger <manuel@rueg.eu>
Co-authored-by: Damien Grisonnet <damien.grisonnet@epita.fr>

k8s-ci-robot committed:
Merge pull request #2176 from mrueg/prep-v2.10.0

chore: Cut v2.10.0

k8s-ci-robot committed:
Merge pull request #2181 from kubernetes/release-2.10

chore: Merge Release 2.10 back to main

mrueg committed:
chore: Bump to kubernetes 1.28

mrueg committed:
chore: Build with go 1.21

opeco17 committed:
feat: introduce Markdown formatter

k8s-ci-robot committed:
Merge pull request #2182 from mrueg/kube-1.28

chore: Bump to kubernetes 1.28

k8s-ci-robot committed:
Merge pull request #2183 from mrueg/go-1.21

chore: Build with go 1.21

dependabot[bot] committed:
build(deps): Bump actions/checkout from 3 to 4

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #2188 from kubernetes/dependabot/github_actions/actions/checkout-4

build(deps): Bump actions/checkout from 3 to 4

mrueg committed:
Fix: Remove FieldSelector from non-namespaced resources

This should resolve the issue with namespace-denylist as discovered in https://github.com/kubernetes/kube-state-metrics/issues/2187

Regression introduced in https://github.com/kubernetes/kube-state-metrics/commit/c3c552821af3432afdaedec607f43ba1efba1be0

Signed-off-by: Manuel Rüger <manuel@rueg.eu>

k8s-ci-robot committed:
Merge pull request #2184 from opeco17/feat/introduce-markdown-formatter

feat: introduce Markdown linter and formatter

dgrisonnet committed:
store: fix missing cluster-scoped resources

Signed-off-by: Damien Grisonnet <dgrisonn@redhat.com>

k8s-ci-robot committed:
Merge pull request #2190 from dgrisonnet/fix-regression

fix: Remove FieldSelector from non-namespaced resources

dependabot[bot] committed:
build(deps): Bump k8s.io/apimachinery from 0.28.1 to 0.28.2

Bumps [k8s.io/apimachinery](https://github.com/kubernetes/apimachinery) from 0.28.1 to 0.28.2.
- [Commits](https://github.com/kubernetes/apimachinery/compare/v0.28.1...v0.28.2)

---
updated-dependencies:
- dependency-name: k8s.io/apimachinery
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #2196 from kubernetes/dependabot/go_modules/k8s.io/apimachinery-0.28.2

build(deps): Bump k8s.io/apimachinery from 0.28.1 to 0.28.2

dependabot[bot] committed:
build(deps): Bump k8s.io/sample-controller from 0.28.1 to 0.28.2

Bumps [k8s.io/sample-controller](https://github.com/kubernetes/sample-controller) from 0.28.1 to 0.28.2.
- [Commits](https://github.com/kubernetes/sample-controller/compare/v0.28.1...v0.28.2)

---
updated-dependencies:
- dependency-name: k8s.io/sample-controller
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #2197 from kubernetes/dependabot/go_modules/k8s.io/sample-controller-0.28.2

build(deps): Bump k8s.io/sample-controller from 0.28.1 to 0.28.2
